src/Views/OperatingSystemView.vala

```diff
@@ -125,17 +125,24 @@ public class About.OperatingSystemView : Gtk.Grid {
 
         var settings_restore_button = new Gtk.Button.with_label (_("Restore Default Settings"));
 
+        var reboot_to_firmware_setup_button = new Gtk.Button.with_label (_("Reboot to Firmware Setup"));
+
         var button_grid = new Gtk.ButtonBox (Gtk.Orientation.HORIZONTAL) {
             hexpand = true,
             layout_style = Gtk.ButtonBoxStyle.END,
             spacing = 6
         };
         button_grid.add (settings_restore_button);
+        button_grid.set_child_secondary (settings_restore_button, true);
+        if (LoginManager.get_instance ().can_reboot_to_firmware_setup ()) {
+            button_grid.add (reboot_to_firmware_setup_button);
```

Comment:
Can we add this on the Firmware page instead of the Operating System page? It seems like that's where it's most relevant